src/azul/terraform.py

```diff
@@ -0,0 +1,29 @@
+import functools
+import itertools
+import json
+import os
+import subprocess
+from typing import (
+    Sequence,
+)
+
+from azul import (
+    require,
+)
+
+
+@functools.lru_cache
```

Comment:
I dislike use of `lru_cache` when there are no parameters that warrant the dict caching employed by `lru_cache`

Convert to method of a singleton class (see AWS in `deployment`) and apply `@cached_property` instead.

Class and instance names should be `Terraform` and `terraform` respectively.